### PR TITLE
🌱 update to go 1.21

### PR DESCRIPTION
**Pull request by kranurag7**
**What type of PR is this?**
/kind feature  

**What this PR does / why we need it**:

This PR updates Go and cluster-api.

Additionally, some small bug fixes for the caphv label handling were added.

The Kubernetes Version was increased to v1.28.6

### CODE REVIEW

**Review by batistein — DISMISSED** [review]

File: controllers/hivelocitymachine_controller.go [hunk 1]
```diff
@@ -246,10 +246,12 @@ func (r *HivelocityMachineReconciler) SetupWithManager(ctx context.Context, mgr
 
 // HivelocityClusterToHivelocityMachines is a handler.ToRequestsFunc to be used to enqeue requests for reconciliation
 // of HivelocityMachines.
-func (r *HivelocityMachineReconciler) HivelocityClusterToHivelocityMachines(ctx context.Context, log logr.Logger) handler.MapFunc {
-	return func(o client.Object) []ctrl.Request {
+func (r *HivelocityMachineReconciler) HivelocityClusterToHivelocityMachines(_ context.Context) handler.MapFunc {
+	return func(ctx context.Context, o client.Object) []ctrl.Request {
 		result := []ctrl.Request{}
 
+		log := log.FromContext(ctx)
```
Comment (janiskemper) [inline controllers/hivelocitymachine_controller.go:253]: two options: either name it "logger" or use another function than log.FromContext. I believe you could use the ctrl packagr well. this way you do an import shadow

File: main.go [hunk 2]
```diff
@@ -64,6 +69,7 @@ var (
 	hivelocityClusterConcurrency int
 	hivelocityMachineConcurrency int
 	logLevel                     string
+	syncPeriod                   time.Duration
```
Comment (janiskemper) [inline main.go:72]: do you have a flag for this?

**Review by janiskemper — CHANGES_REQUESTED** [review]

**Review by janiskemper — DISMISSED** [review]

**Review by guettli — CHANGES_REQUESTED** [review]
First we need to check that via Tilt. Then we can solve the e2e tests.

**Review by guettli — DISMISSED** [review]

**Comment by guettli** [comment]
we splitted this big PR into two:

https://github.com/hivelocity/cluster-api-provider-hivelocity/pull/139

https://github.com/hivelocity/cluster-api-provider-hivelocity/pull/140

**Pull request closed without merging**